src/sentry/snuba/metrics/query_builder.py

```diff
@@ -566,14 +578,11 @@ def _extract_data(self, data, groups):
     def translate_results(self):
         groups = {}
         for entity, subresults in self._results.items():
-            totals = subresults["totals"]["data"]
-            for data in totals:
-                self._extract_data(data, groups)
-
-            if "series" in subresults:
-                series = subresults["series"]["data"]
-                for data in series:
-                    self._extract_data(data, groups)
+            for k in "totals", "series":
+                if k in subresults:
+                    totals = subresults[k]["data"]
```

Comment:
nit: I would name this something other than `totals` cause it is a bit confusing

Reply:
done